File: examples/python/single-responsibility-principle/bad.py
import sys
import os

sys.path.append(os.path.abspath(os.path.join(
    os.path.dirname(__file__), '..')
))

import math
from matplotlib.pyplot import isinteractive
from models.Circle import Circle
from models.Square import Square

class AreaCalculator:
    def __init__(self, shapes: list):
        self.shapes = shapes

    def sum(self) -> float:
        
        sum_area = 0
        
        for shape in self.shapes:
            if isinstance(shape, Square):
                sum_area += shape.length**2
            if isinstance(shape, Circle):
                sum_area += math.pi*(shape.radius**2)
        
        return sum_area

    def output(self) -> str:
        return f"Sum of the areas of provided shapes: {self.sum():.2f}"



'''
Here is an example with a collection of three shapes:

a circle with a radius of 2
a square with a length of 5
a second square with a length of 6
'''

shapes = [
  Circle(2),
  Square(5),
  Square(6),
]

areas = AreaCalculator(shapes)

print(areas.output())

File: examples/csharp/SingleResponsibilityPrinciple/Bad.cs
using System;

namespace SRP.Bad
{
    public class Circle
    {
        public int Radius { get; private set; }

        public Circle(int radius)
        {
            this.Radius = radius;
        }
    }

    public class Square
    {
        public int Length { get; private set; }

        public Square(int length)
        {
            this.Length = length;
        }
    }

    class AreaCalculator
    {
        private List<object> shapes = null;
        public AreaCalculator(List<object> shapes)
        {
            this.shapes = shapes;
        }

        public double Sum()
        {
            double sumArea = 0.0;

            if (this.shapes == null)
                return 0;
            
            foreach (var shape in this.shapes)
            {
                if (shape.GetType() == typeof(Square))
                    sumArea += Math.Pow(((Square)shape).Length, 2);

                if (shape.GetType() == typeof(Circle))
                    sumArea += Math.PI * Math.Pow(((Circle)shape).Radius, 2);
            }
            
            return sumArea;
        }

        public string Output()
        {
            return String.Format("Sum of the areas of provided shapes: {0:0.00}", this.Sum());
        }
    }
}
